### ISSUE TITLE
Allow pull request author to "Request Changes" themselves

### ISSUE THREAD

**Issue by adamdecaf** [dear-github/dear-github#332]
Often a pull request (PR) author might share a branch in PR form to get feedback from the team or to review at a later point. It would be helpful for the author to be able to "Request Changes" on themselves. The original author should be able to signal not to merge if desired. 

This also applies when a PR changes ownership, if an employee leaves for example.

**Comment by clarkbw** [comment]
Would a Work In Progress type system help here?  Something like #191

**Comment by adamdecaf** [comment]
I'd call WIP a special case of "not ready to merge". Likely the most common though.

**Comment by IronSean** [comment]
I understand not having this behaviour for Open Source, but for some private projects I sometimes get into the situation where a team member is gone for the day and didn't open a PR. If I want to review it and use the github tools to provide feedback, I end up being the PR author and therefore can't make use of these tools, request changes, or approve anymore.

**Comment by abhijit-hota** [comment]
+1 on @IronSean. I just completed a very detailed review of a PR that I made from a teammate's branch only to find out I cannot request changes. While it's only fair and this be the case for public OS projects, it would be very useful in private personal or team projects.

I think the same pattern as `Assign to` in issues can be followed. Something like `Request changes to:` and then a dropdown of organization members. With suggestions of repo owners, and commit authors, etc.

**Comment by BrettSterling-SpectraLogic** [comment]
+1 - after writing the code, if the author discovers a bug and wants the other reviewers to know about it, this would be a very beneficial feature.  Otherwise, if there is automatic merging on the project, the only option the author has currently is to cancel the PR.  This is a significant amount of extra work, especially if others have commented already.

**Comment by hXtreme** [comment]
Hit this same issue, team-mate didn't create a pr so I open it, write a review and notice that I can't actually request for changes.

**Comment by NogginBops** [comment]
I want to leave a request changes review to indicate that my own PR needs changes related to another PR (so that I don't merge it prematurely), and I'm not going to be working on this PR for a little while so requesting changes seemed like a good idea to remind myself that there was work left to do. 
But apparently I'm not allowed to...

**Comment by tiansu-yu-idealo** [comment]
Upvote for this issue.

**Comment by AlanSl** [comment]
Same request please. In my case:

- my PR **is** ready for review and I want colleagues to review it, so setting it to "draft" isn't appropriate,
- but there is one very localised crucial and non-obvious problem with it that must be fixed before it is merged
- This problem should block merging until it is fixed (and there is a danger that colleagues in a different timezone could miss my comment flagging the issue, approve, and merge during hours I'm not active), but shouldn't deter the rest of the PR from being reviewed

So "request changes" is exactly the right tool for my scenario, because it flags the problem, blocks merge, but continues to allow and encourage the rest of the PR to be reviewed; but I'm not allowed to use it.

**Comment by amethystmarie** [comment]
Definitely a problem when multiple people are collaborating on a single PR. The person that opened the PR can't request changes for parts of the code someone else worked on.

**Comment by zakimirza88** [comment]
PR authors must be able to request changes in a review, mainly because the draft PR comprises commits from multiple contributors including reviewers and the PR author themselves, and not allowing that is unnecessarily restrictive.

**Comment by naman-gupta99** [comment]
+1, this also helps in keeping the changes that needs to be done before merging in master but not while testing.

**Comment by Convolva** [comment]
+1

**Comment by ivan-moto** [comment]
+1

**Comment by aamirazad** [comment]
+1

**Comment by Amer-Jabar** [comment]
+1

**Comment by emmett2020** [comment]
+1

**Comment by AltarBoii** [comment]
+1

**Comment by joelclimbsthings** [comment]
+1

**Comment by pguillory** [comment]
+1

**Comment by elitalpa** [comment]
+1

**Comment by fybx** [comment]
+1

**Comment by wicresoft-cv** [comment]
+1

**Comment by shammlo** [comment]
+1

**Comment by mariela21180** [comment]
+1

**Comment by teroqim** [comment]
+1

**Comment by LockedThread** [comment]
+1

**Comment by F4CTE** [comment]
+1

**Comment by wgcorrea** [comment]
+1

**Comment by MasterDash5** [comment]
+1

**Comment by slayton** [comment]
In the era when Agents are writing code and opening PRS for us its not uncommon for an org to setup the process so that the agent opens a PR on behalf of the engineer guiding the work. It would make sense if the engineer could then use the "Request Changes" review to signal back to the bot that further work is needed.

**Comment by kenmclennan** [comment]
> In the era when Agents are writing code and opening PRS for us its not uncommon for an org to setup the process so that the agent opens a PR on behalf of the engineer guiding the work. It would make sense if the engineer could then use the "Request Changes" review to signal back to the bot that further work is needed.

A thousand times this!

**Comment by sun** [comment]
+1 — My AI agent is creating PRs on behalf of me and I need to review them.  Not being able to request changes complicates the process.
